### ISSUE TITLE
Add relevant CSS classes to stickied, locked, and maybe subscribed discussions

### ISSUE THREAD

**Issue by askvortsov1** [flarum/core#2128]
## Feature Request

**Is your feature request related to a problem? Please describe.**
It's difficult for forum admins to add custom CSS to discussions that are stickied only, such as a highlighted background. The same applies to locked and maybe subscribed discussions.
Relevant: https://discuss.flarum.org/d/23486-is-it-possible-to-change-background-color-of-sticky-post

**Describe the solution you'd like**
A 'discussion-sticky' (or similar) css class should be added to sticked discussions, and the same should be done for locked and subscribed ones.

**Justify why this feature belongs in Flarum's core, rather than in a third-party extension**
This will allow forum admins to easily add custom CSS to sticky discussions only.

**Comment by SychO9** [comment]
Not an urgent issue, but great to have to improve the theming experience.

**Comment by davwheat** [comment]
Maybe we should consider using data attributes for this instead?

`data-discussion-type="sticky subscribed"` would allow for CSS to use `[data-discussion-type*="subscribed"]` to match all subscribed discussions?

**Comment by SychO9** [comment]
Since the purpose is styling, modifier class names are better and more consistent with the code base.

**Comment by davwheat** [comment]
I'm aware that it's more consistent, but I think that at some point we should start moving towards data attributes.

Attributes like these are more commonly used for storing data related to the component/element, which I think is a perfect fit for this situation.

We already use `data-type`, `data-number`, etc on post stream items. Why is that so different to this?

```html
<div class="Discussion Discussion--stickied Discussion--locked">

<div class="Discussion" data-modifiers="stickied locked">
```

To me both of these are equally readable.

**Comment by SychO9** [comment]
> Attributes like these are more commonly used for storing data related to the component/element, which I think is a perfect fit for this situation.
>
> We already use data-type, data-number, etc on post stream items. Why is that so different to this?

What's different is those were not introduced for the purpose of styling, but for JS manipulation. And there doesn't seem to be a valid reason for why we should drop modifier class names in favor of data attributes and therefore introduce more overly specified selectors in our styles.

Here's another proposition instead, we can introduce data attributes, while still use modifier class names for CSS selectors, even though I generally prefer to avoid introducing anything unless there's a clear need for it.

tldr; I really like the convention we're following and would rather we stick to it (https://github.com/suitcss/suit/tree/master/doc).

(*Also ref previous discussion of this: https://github.com/flarum/core/pull/2797#discussion_r616759240*)
